iroh-net/src/endpoint.rs

```diff
@@ -1138,15 +1137,345 @@ impl Future for Connecting {
             Poll::Ready(Err(err)) => Poll::Ready(Err(err)),
             Poll::Ready(Ok(conn)) => {
                 try_send_rtt_msg(&conn, this.ep);
-                Poll::Ready(Ok(conn))
+                Poll::Ready(Ok(Connection { inner: conn }))
             }
         }
     }
 }
 
-/// Extract the [`PublicKey`] from the peer's TLS certificate.
-// TODO: make this a method now
-pub fn get_remote_node_id(connection: &quinn::Connection) -> Result<PublicKey> {
+/// A QUIC connection.
+///
+/// If all references to a connection (including every clone of the Connection handle,
+/// streams of incoming streams, and the various stream types) have been dropped, then the
+/// connection will be automatically closed with an error_code of 0 and an empty reason. You
+/// can also close the connection explicitly by calling Connection::close().
+///
+/// Closing the connection immediately abandons efforts to deliver data to the peer. Upon
+/// receiving CONNECTION_CLOSE the peer may drop any stream data not yet delivered to the
+/// application. Connection::close() describes in more detail how to gracefully close a
+/// connection without losing application data.
+///
+/// May be cloned to obtain another handle to the same connection.
+#[derive(Debug, Clone)]
+pub struct Connection {
```

Comment:
Can we have #[repr(transparent)] on this? Then we could use ref_cast to get a &quinn::Connection from a &Connection as a (possibly feature flagged) compat feature.

We might or might not do this, but the repr(transparent) certainly does not do any harm unless we plan to have additional stuff here...